lib/ukcpio/cpio.c

```diff
@@ -58,6 +58,37 @@ int uk_syscall_r_utime(const char *, const struct utimbuf *);
 int uk_syscall_r_mkdir(const char *, mode_t);
 int uk_syscall_r_symlink(const char *, const char *);
 int uk_syscall_r_stat(const char *, struct stat *);
+int uk_syscall_r_unlinkat(int, const char *, int);
+int uk_syscall_r_rename(const char *, const char *);
+
+static int
+try_remove(const char *path)
+{
+	int r;
+
+	uk_pr_info("Trying to unlink %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, 0);
+	if (!r || r == -ENOENT)
+		return 0;
+
+	uk_pr_info("Trying to rmdir %s\n", path);
+	r = uk_syscall_r_unlinkat(AT_FDCWD, path, AT_REMOVEDIR);
+	if (!r || r == -ENOENT) {
+		return 0;
```

Comment:
The static code checker unfortunately can't tell that the else block is there to prevent a `PATH_MAX` sized object from coming into scope unless certain conditions are met, you need human reasoning for that. Compiler optimizations _may_ be clever enough to avoid allocating `newpath` unless on the branch making use of it, but it's nowhere near guaranteed.

Reply:
Ok so the only reason you keep this is for optimizations to potentially and undeterministically be able to avoid allocating a buffer on the stack?

Reply:
I think you misunderstand, the "undeterministic" optimizations would happen if I were to remove the else and declare `newpath` in the function scope. The way it is now (with "redundant" else) it is guaranteed by the C standard that `newpath` will only be allocated when it comes in scope.